src/lib/terser-options.js

```diff
@@ -141,7 +141,15 @@ const options = `// edit terser options
   rename: {},
 }`;
 
-/* eslint-disable-next-line no-eval */
-export const evalOptions = (opts) => eval(`(${opts||options})`)
+export const evalOptions = (opts) => {
+  opts = opts || options;
+  // Strip line comments
+  opts = opts.replace(/\/\/.*/g, '');
+  // Trim trailing commas
+  opts = opts.replace(/,\s*([\]}])/g, '$1');
+  // Quote property names
+  opts = opts.replace(/^\s*(\w+):/gm, '"$1":');
+  return JSON.parse(opts);
+}
```

Comment:
I see why this is something we need to do (avoiding a self xss and whatnot), however there are some Terser options that are regexps. For instance you can use `/^_/` to select properties started with _ to mangle :/

There are a few options that take functions too (which would be a nightmare) so I'm not mentioning that part.

Reply:
I think it's possible to avoid this by disabling sharing for options that include RegExps and functions. Then you can do eval freely, and only do JSON.parse in loadState